ref/vk/r_speeds.c

```diff
@@ -706,35 +711,98 @@ static const char *getMetricTypeName(r_speeds_metric_type_t type) {
 	return "UNKNOWN";
 }
 
-// Ideally, we'd just autocomplete the r_speeds_graphs cvar/cmd.
-// However, autocompletion is not exposed to the renderer. It is completely internal to the engine, see con_utils.c, var cmd_list.
-static void listMetrics( void ) {
-	if (gEngine.Cmd_Argc() > 1) {
-		Q_strncpy(g_speeds.frame.list_metrics_filter, gEngine.Cmd_Argv(1), sizeof(g_speeds.frame.list_metrics_filter));
-	} else {
-		g_speeds.frame.list_metrics_filter[0] = '\0';
+// Returns pointer to filename in filepath string.
+// Maybe function like this should be inside filesystem?
+// Examples:
+// on Windows: C:\Users\User\xash3d-fwgs\ref\vk\vk_rtx.c  ->  vk_rtx.c
+// on Linux:   /home/user/xash3d-fwgs/ref/vk/vk_rtx.c     ->  vk.rtx.c (imaginary example, not tested)
+static const char *get_filename_from_filepath( const char *filepath ) {
+	size_t length = Q_strlen( filepath );
+	int cursor = length - 1;
+	while ( cursor > 0 ) {
+		char c = filepath[cursor];
+		if ( c == '/' || c == '\\' ) {
+			// Advance by 1 char to skip the folder delimiter symbol itself, but
+			// make sure that we are not exceeding the length.
+			if ( cursor < length )
+				cursor += 1;
+
+			return &filepath[cursor];
+		}
+		cursor -= 1;
 	}
 
-	g_speeds.frame.list_metrics = true;
+	return NULL;
```

Comment:
Если ему дали односимвольного чувака, пусть он его самого и вернёт:
```suggestion
	return filepath;
```